src/command_modules/azure-cli-interactive/azclishell/telemetry.py

```diff
@@ -84,5 +103,9 @@ def scrub(text):
 
 
 TC = Telemetry(INSTRUMENTATION_KEY)
-enable(INSTRUMENTATION_KEY)
-my_context(TC)
+
+
+if __name__ == '__main__':
+    # If user doesn't agree to upload telemetry, this scripts won't be executed. The caller should control.
+    TC.track_event('Run', json.loads(sys.argv[1]))
```

Comment:
track_event is the super class version, _track_event is for the interactive mode which generates the thread